soc/arm/nxp_imx/rt5xx/soc.c

```diff
@@ -458,12 +462,20 @@ void __weak imxrt_pre_init_display_interface(void)
 	 * We set the divider of the PFD3 output of the SYSPLL, which has a
 	 * fixed multiplied of 18, and use this output frequency for the DPHY.
 	 */
+
+#ifdef CONFIG_MIPI_DPHY_CLK_SRC_AUX1_PLL
+	/* Note: AUX1 PLL clock is system pll clock * 18 / pfd.
+	 * system pll clock is configured at 528MHz by default.
+	 */
 	CLOCK_AttachClk(kAUX1_PLL_to_MIPI_DPHY_CLK);
 	CLOCK_InitSysPfd(kCLOCK_Pfd3,
 		((CLOCK_GetSysPllFreq() * 18ull) /
 		((unsigned long long)(DT_PROP(DT_NODELABEL(mipi_dsi), phy_clock)))));
 	CLOCK_SetClkDiv(kCLOCK_DivDphyClk, 1);
-
+#elif defined(CONFIG_MIPI_DPHY_CLK_SRC_FRO)
+	CLOCK_AttachClk(kFRO_DIV1_to_MIPI_DPHY_CLK);
+	CLOCK_SetClkDiv(kCLOCK_DivDphyClk, 1);
```

Comment:
For the other conditional case, we do some basic calculations to determine the clock divider, based on the `phy-clock` property.

We don't seem to have a function to determine the FRO clock frequency within the SDK, but there is a constant defined: [CLK_FRO_CLK](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/mcux-sdk/devices/MIMXRT595S/system_MIMXRT595S_cm33.h#L88). Could we do something like the following:

`CLOCK_SetClkDiv(kCLOCK_DivDphyClk, (CLK_FRO_CLK / DT_PROP(DT_NODELABEL(mipi_dsi), phy_clock))));`

I'm not sure if you'd find this valuable, I just want to preserve the functionality of the `phy-clock` property if we can.

Reply:
thanks @danieldegrasse , updated as you requested